smallrye-reactive-messaging-amqp/src/test/java/io/smallrye/reactive/messaging/amqp/AmqpFailureHandlerTest.java

```diff
@@ -131,6 +140,46 @@ public void testRejectStrategy() {
         assertThat(isAmqpConnectorReady(connector)).isTrue();
     }
 
+    @Test
+    public void testModifiedFailedStrategy() {
+        getModifiedFailedConfig();
+        MyReceiverBeanRecovering bean = deployRecovering();
+        AtomicInteger counter = new AtomicInteger();
+        AmqpConnector connector = container.getBeanManager().createInstance().select(AmqpConnector.class,
+                ConnectorLiteral.of(AmqpConnector.CONNECTOR_NAME)).get();
+        await().until(() -> isAmqpConnectorReady(connector));
+        await().until(() -> isAmqpConnectorAlive(connector));
+
+        usage.produceTenIntegers("modified-failed", counter::getAndIncrement);
+
+        await().atMost(2, TimeUnit.MINUTES).until(() -> bean.list().size() >= 10);
+        // All messages should not have been received + 3, 6 and 9 are redelivered
+        assertThat(bean.list()).containsExactly(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 3, 6, 9);
+
+        assertThat(isAmqpConnectorAlive(connector)).isTrue();
+        assertThat(isAmqpConnectorReady(connector)).isTrue();
+    }
+
+    @Test
+    public void testModifiedFailedUndeliverableHereStrategy() {
+        getModifiedFailedUndeliverableConfig();
+        MyReceiverBeanRecovering bean = deployRecovering();
+        AtomicInteger counter = new AtomicInteger();
+        AmqpConnector connector = container.getBeanManager().createInstance().select(AmqpConnector.class,
+                ConnectorLiteral.of(AmqpConnector.CONNECTOR_NAME)).get();
+        await().until(() -> isAmqpConnectorReady(connector));
+        await().until(() -> isAmqpConnectorAlive(connector));
+
+        usage.produceTenIntegers("modified-failed-undeliverable-here", counter::getAndIncrement);
+
+        await().atMost(2, TimeUnit.MINUTES).until(() -> bean.list().size() >= 10);
+        // All messages should not have been received, 3 6 and 9 are NOT redelivered
```

Comment:
Should have been received?

Reply:
yes, we capture the list before the failure.